tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/HttpTelemetryTests.cs

```diff
@@ -64,63 +69,171 @@ public async Task AcquireTokenInteractive_UnknownError_FollowedByASuccessfulToke
             await CreateFailedAndThenSuccessfulResponseToVerifyErrorCodesInHttpTelemetryDataAsync(MsalError.UnknownError).ConfigureAwait(false);
         }
 
+        [TestMethod]
+        public void CheckCurrentHttpTelemetryHeaderContent()
+        {
+            const string csvString = "2|,1|";
+            HttpTelemetryContent httpTelemetryContent = CreateHttpTelemetryContent();
+            Assert.AreEqual(csvString, httpTelemetryContent.GetCsvAsCurrent());
+        }
+
+        [TestMethod]
+        public void CheckPreviousHttpTelemetryHeaderContentWithAuthFailed()
+        {
+            const string csvString =
+                "2|0|" +
+                ",1005,1005" +
+                ",d3adb33f-c0de-ed0c-c0de-deadb33fc0d3,0a69ff54-3e00-4244-a6c8-09ccc1efa707|" +
+                ",authentication_failed|";
+
+            HttpTelemetryContent httpTelemetryContent = CreateHttpTelemetryContent();
+            List<string> errorCodes = new List<string>(new string[] { MsalError.AuthenticationFailed });
+            PopulateHttpTelemetryContent(errorCodes);
+
+            Assert.AreEqual(csvString, httpTelemetryContent.GetCsvAsPrevious(0));
+        }
+
+        [TestMethod]
+        public void CheckPreviousHttpTelemetryHeaderContentWithTwoFailures()
+        {
+            const string csvString =
+                "2|0|" +
+                ",1005,1005" +
+                ",d3adb33f-c0de-ed0c-c0de-deadb33fc0d3,0a69ff54-3e00-4244-a6c8-09ccc1efa707|" +
+                ",authentication_failed,user_mismatch|";
+
+            HttpTelemetryContent httpTelemetryContent = CreateHttpTelemetryContent();
+            List<string> errorCodes = new List<string>(new string[] { MsalError.AuthenticationFailed, MsalError.UserMismatch });
+            PopulateHttpTelemetryContent(errorCodes);
+
+            Assert.AreEqual(csvString, httpTelemetryContent.GetCsvAsPrevious(0));
+        }
+
+        [TestMethod]
+        public void CheckPreviousHttpTelemetryHeaderContentWithThreeFailures()
+        {
+            const string csvString =
+                "2|1|" +
+                ",1005,1005,1005" +
+                ",d3adb33f-c0de-ed0c-c0de-deadb33fc0d3,0a69ff54-3e00-4244-a6c8-09ccc1efa707,b1e662cf-8efb-4e13-b89a-71e845bbb62f|" +
+                ",authentication_failed,user_mismatch,invalid_grant|";
+
+            HttpTelemetryContent httpTelemetryContent = CreateHttpTelemetryContent();
+            List<string> errorCodes = new List<string>(new string[] { MsalError.AuthenticationFailed, MsalError.UserMismatch, MsalError.InvalidGrantError });
+            PopulateHttpTelemetryContent(errorCodes);
+            HttpTelemetryContent.ApiId.Add("1005");
+            HttpTelemetryContent.CorrelationId.Add("b1e662cf-8efb-4e13-b89a-71e845bbb62f");
+
+            Assert.AreEqual(csvString, httpTelemetryContent.GetCsvAsPrevious(1));
+        }
+
+        [TestMethod]
```

Comment:
These tests are ok, but we need to have higher level tests that check this. We need to assert on the telemetry headers on E2E tests, either unit tests or integration tests. 

I'd expect to have a test that does smth like: 

1. Acquire Token Interactive (or U/P or whatever)
-> check telemetry 
2. Acquire Token Silent
-> check telemetry
3. Acquire Token Silent again, with an expired AT
-> check telemetry